packages/interactivity/src/proxies/signals.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import {
+	computed,
+	signal,
+	batch,
+	type Signal,
+	type ReadonlySignal,
+} from '@preact/signals-core';
+
+/**
+ * Internal dependencies
+ */
+import { getProxyNs } from './registry';
+import { getScope, setNamespace, resetNamespace } from '../hooks';
+import { withScope } from '../utils';
+
+/**
+ * Identifier for property computeds not associated to any scope.
+ */
+const NO_SCOPE = Symbol();
+
+/**
+ * Structure that manages reactivity for a property in a state object. It uses
+ * signals to keep track of property value or getter modifications.
+ */
+export class PropSignal {
```

Comment:
I wonder if we should change this name. Have you thought of any other options?

Reply:
Naming is tricky. 😅

I thought of other options, but this one was the most appropriate (in my opinion). I'm open to suggestions, though! 😄

Reply:
It's internal, so let's leave it like this for now. We can change it in the future if we come up with something better 🙂